library/rsa_alt_helpers.c

```diff
@@ -126,7 +126,7 @@ int mbedtls_rsa_deduce_primes(mbedtls_mpi const *N,
     }
 
     for (; attempt < num_primes; ++attempt) {
-        mbedtls_mpi_lset(&K, primes[attempt]);
+        MBEDTLS_MPI_CHK(mbedtls_mpi_lset(&K, primes[attempt]));
```

Comment:
Consequence of failure: a failure can only happen on the first iteration, after that `K` has enough room since all the values fit in one word and `K` doesn't shrink. The first iteration uses K=0. I'm not sure what the possible numerical consequences here, but I don't see any potential for memory corruption, just an incorrect result when reconstructing the CRT parameters of an RSA key from the private exponent (confirmed experimentally by commenting out this line and running the unit tests).